waspc/src/Wasp.hs

```diff
@@ -57,6 +59,10 @@ data Wasp = Wasp
     { waspElements :: [WaspElement]
     , waspJsImports :: [JsImport]
     , externalCodeFiles :: [ExternalCode.File]
+
+    -- NOTE(matija): This is for demo purposes only. This option should come as
```

Comment:
Why not put it immeditaely as compile option?

Reply:
The problem was CompileOptions is not wired everywhere, while Wasp is in a majority of places. So I wanted to avoid adding everywhere CompileOptions. We'd probably need to do some refactoring for this as you said, maybe some ReaderT or sth. But ideally it would be a compile option for sure, it would be super cool to be able to specify it as a flag --xray-mode=on :D.

Reply:
Ok true, we need smth in style of ReaderT I guess hm! Cool cool cool.